### PR TITLE
Integrate API calls to fetch user self data and calendar ID to get user specific events

### PR DESCRIPTION
**Pull request by harshith-venkatesh**
### What is the change?

Added API integration for fetching user self and calendar ID data

### Is it bug?

- Steps to repro
- Expected
- Actual

### \*Dev Tested?

- [x] Yes
- [ ] No

#### Platforms

- [x] Web

#### Browsers

- [x] Chrome
- [x] Safari
- [x] Firefox

### Before / After Change Screenshots

> For visual or interaction changes. Can be video / screenshot.

### CODE REVIEW

**Comment by harshith-venkatesh** [comment]
Dependent on #163

**Comment by iRohitGaur** [comment]
Screenshot 2023-06-26 at 10 03 18

Should be comma separated

**Comment by harshith-venkatesh** [comment]
Will handle navbar data flickering issue and comma separated data issue in this PR itself

File: app/routes/index.tsx [hunk 1]
```diff
@@ -24,7 +24,10 @@ export const loader: LoaderFunction = async ({ request }) => {
   const endTime = dayjs().add(1, 'months').endOf('month').unix() * 1000;
 
   try {
-    const response = await axios.get(getEvents(process.env.API_HOST ?? '', startTime, endTime), {
+    const {data : { username}} = await axios.get(getUserSelfData(process.env.API_HOST ?? ''));
```
[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply app/routes/index.tsx:27]: I am confused, Are you calling the current code bad idea?

File: app/routes/index.tsx [hunk 2]
```diff
@@ -35,7 +38,7 @@ export const loader: LoaderFunction = async ({ request }) => {
     return { events: null, ENV: baseUrls, error };
   }
 };
-export const unstableShouldReload: ShouldReloadFunction = () => false;
+export const unstableShouldReload: ShouldRevalidateFunction = () => false;
```
[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply app/routes/index.tsx:41]: After the upgrade of remix for remix testing ShouldReloadFunction had been deprecated, this is replaced with current function

File: app/components/common/eventCard/index.test.tsx [hunk 3]
```diff
@@ -1,108 +1,65 @@
...
+  expect(eventCardLocation).toBeInTheDocument();
+});
 
-  it('should render the correct month', () => {
-    const { getByText } = render(<EventCard {...dummyevent1} />);
-    const monthText = month;
-    const eventMonth = getByText(monthText);
-    expect(eventMonth).toBeInTheDocument();
-  });
+// TODO: fix issue useHref() may be used only in the context of a <Router> component.
```
[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply app/components/common/eventCard/index.test.tsx]: Have created a separate task for the same
Would you like to take it up?

File: app/components/common/eventCard/index.tsx [hunk 4]
```diff
@@ -1,39 +1,95 @@
...
-            <div className="text-black font-medium text-base">{date}</div>
-            <div className="text-black font-medium text-sm">{month}</div>
-          </div>
+const EventCard = ({
+  title,
+  start,
+  end,
+  attendees,
+  location,
+  meetLink
```
[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply app/components/common/eventCard/index.tsx]: Will update the changes

File: app/components/common/eventCard/index.tsx [hunk 5]
```diff
@@ -1,39 +1,95 @@
...
+          )}
+          {!attendees && !location && meetLink && (
+            <Link
+              target={'blank'}
+              to={meetLink}
+              data-testid="event-card-meet"
+              className="w-full grid grid-cols-[auto,1fr] items-center gap-1"
+            >
+              <div className="w-3 h-4 flex items-center mr-1 pt-0.5">
+                <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 87.5 72">
```
[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply app/components/common/eventCard/index.tsx]: Will update the change

File: app/routes/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply app/routes/index.tsx:41]: https://remix.run/docs/en/main/route/should-revalidate

File: app/components/common/eventCard/index.test.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply app/components/common/eventCard/index.test.tsx]: https://github.com/Real-Dev-Squad/website-calendar/issues/164

[the remark this answers is not in the document]
Comment (isVivek99) [reply app/components/common/eventCard/index.test.tsx]: yes, I can work on that

File: app/routes/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (isVivek99) [reply app/routes/index.tsx:27]: nope, not ur, my code

File: app/routes/index.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (isVivek99) [reply app/routes/index.tsx:41]: have we upgraded remix's version?

[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply app/routes/index.tsx:41]: Yes, while I was testing for adding remix stubs

File: app/constants/userEvents.constants.ts [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+      description:
+        'Lorem ipsum dolor sit amet, consectetur adipiscing elit ut aliquam, purus sit amet luctus venenatis, lectus magna fringilla urna, porttitor rhoncus dolor purus non enim praesent elementum facilisis leo, vel.',
+      location: 'Nashville, Tennessee. U.S.',
+      start: dayjs().add(1, 'hour').toDate(),
+      end: dayjs().add(2, 'hours').toDate(),
+      visibility: 'private',
+      attendees: [{ attendee: { email: 'a@b.c' } }, { attendee: { email: 'alpha@beta.gamma' } }],
+      onlineEventLink: 'https://github.com/Real-Dev-Squad/website-calendar'
+    },
+  ];
```
Comment (iRohitGaur) [inline app/constants/userEvents.constants.ts:18]: We already have events in the `events.data.ts` file

**Review by iRohitGaur — APPROVED** [review]

**Pull request merged**